### PR TITLE
Added tests, then added fixes for those tests

### PR DESCRIPTION
**Pull request by mistergone**
This PR updates format-usd to 0.2.0. No functionality has changed, but there are several fixes.

Basically, while looking over this module for use in my next project, I realized the tests weren't covering negative dollar values, and that the code probably wasn't handling them correctly. I added tests for negatives, and I was right, so I fixed the code to handle negatives.

While I was doing that, I also overhauled the code for verbosity and made it much closer to our [front-end standards](https://github.com/cfpb/front-end). At @ascott1's suggestion, I moved some functionality into separate `./lib` files (and added tests for those functions). Then I fixed the code so that it could handle strings with multiple periods (by assuming the first period is the intended decimal place).

## Additions
- Extra code and comments to make the code clearer and more verbose

## Changes
- Variable names, to make them clearer
- Some functionality separate into functions and placed in `./lib/[some-function].js`

## Testing
- Tests added for negative numbers
- Tests added for `comma-separate-number.js` and `handle-string-input.js`

## Review
- @contolini @ascott1 and anyone else who's interested!

## Checklist
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [x] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices 
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

## Thanks
Thanks to @ascott1 for reviewing the commits and giving feedback!

## Animated GIF
Me when I remove ternary operators:
celebrate!

### CODE REVIEW

**Comment by mistergone** [comment]
> [ ] Placeholder code is flagged
> [ ] Visually tested in supported browsers and devices
> [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)

These parts of the checklist aren't really relevant, but rather than just remove them, I wanted to note that they're not really relevant. :smile: :octopus:

File: test/format-usd-tests.js [hunk 1]
```diff
@@ -72,8 +74,64 @@ exports.formatUSD = {
...
+    test.done();
   }
 };
+
+exports.commaSeparate = {
+  setUp: function(done) {
+    // setup here
+    done();
+  },
+  'Comma separte a numeric string': function(test) {
```
Comment (contolini) [inline test/format-usd-tests.js]: separte -> separate

**Comment by contolini** [comment]
Wow it's comical how illegible this codebase was before this PR. Nice work!

**Comment by mistergone** [comment]
Merged! About to publish!

**Comment by ascott1** [comment]
Big improvements all around.

**Pull request merged**